src/iceberg/schema.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+/// \file iceberg/schema.h
+/// Schemas for Iceberg tables.
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include "iceberg/iceberg_export.h"
+#include "iceberg/type.h"
+
+namespace iceberg {
+
+/// \brief A schema for a Table.
+///
+/// A schema is a list of typed columns, along with a unique integer ID.  A
+/// Table may have different schemas over its lifetime due to schema
+/// evolution.
+class ICEBERG_EXPORT Schema : public StructType {
+ public:
+  Schema(int32_t schema_id, std::vector<Field> fields);
+
+  /// \brief Get the schema ID.
+  ///
+  /// Schemas are identified by a unique ID for the purposes of schema
+  /// evolution.
+  [[nodiscard]] int32_t schema_id() const;
+
+  /// \brief Get a user-readable string representation of the schema.
+  [[nodiscard]] std::string ToString() const;
+
+  /// \brief Compare two schemas for equality.
+  [[nodiscard]] bool Equals(const Schema& other) const;
```

Comment:
If we have operators for equality, can't we make the Equals() function private? I don't see the reason to expose both.

Reply:
It may be useful to add overloads later (e.g. if we want to ignore metadata or IDs)

Reply:
Then we can extend this later. I believe in adding what is required ATM, and if there is a need later, we can still extend.
Also, we are defining a public interface here, should keep it as simple as possible. Exposing the Equals method seems a redundancy for me ATM.